src/index.js

```diff
@@ -15,9 +18,28 @@ async function run() {
     core.getInput('stale-days')
   )
 
-  const commitMessageLines = Number(core.getInput('commit-messages-lines'))
+  const {
+    isClosing,
+    isNotifyReleaseIssue,
+    stateClosedNotPlanned,
+    issueNumber,
+  } = getClosingIssueDetails(context)
 
-  await runAction(token, notifyAfter, commitMessageLines)
+  if (!isClosing) {
```

Comment:
I would prefer if we could change this logic a little to make it more abstract and leak less details. We don't really care much if we're closing "any issue" here, we only care if we're closing _our_ issue as _not planned_. Therefore, I would handle this logic in this way (pseudo code):

```
const isSnoozing = getIsSnoozingIssue(...) // will return true if we're closing our issue as not planned

if(isSnoozing) {
  return addSnoozingComment(...)
}

// continue with the rest of the action
```

this will simplify the logic by avoiding to expose unnecessary details we don't care about

Reply:
Refactored the code as requested @simoneb . I needed to add the `isClosing` check as well, otherwise, if the issue is closing as `complete` the action will run normally and creates another issue.

I've also re-run E2E tests successfully, details are below.

<details>
<summary>Test screenshots:</summary>

1. run manually the action
<img width="1043" alt="image" src="https://user-images.githubusercontent.com/6604111/215746099-bd03d1b0-bac2-4c74-b8d2-15be25fe6022.png">

2. run the action after a release has been published
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/6604111/215747614-9758c4f4-11da-4e34-b00f-190a1ba39ffd.png">

3. close the issue as complete
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/6604111/215747316-76f53077-92f3-4006-9ac2-d37e310599e2.png">

4. close the issue as not_planned
<img width="1049" alt="image" src="https://user-images.githubusercontent.com/6604111/215748011-d3147b9e-902b-468e-b1a8-7f18e5556c9e.png">

<img width="926" alt="image" src="https://user-images.githubusercontent.com/6604111/215747925-3e2da5b4-7ba2-44ab-abe0-3ac4c87fbd3e.png">

</details>